Classes/Domain/Repository/AbstractDemandRepository.php

```diff
@@ -151,13 +151,13 @@ protected function addOrderings(QueryBuilder $queryBuilder, DemandInterface $dem
     /**
      * Fire demand event.
      *
-     * @param string $name
      * @param DemandInterface $demand
      * @param QueryBuilder $queryBuilder
      */
-    protected function fireDemandEvent(string $name, DemandInterface $demand, QueryBuilder $queryBuilder): void
+    protected function fireDemandEvent(DemandInterface $demand, QueryBuilder $queryBuilder): void
     {
-        $event = GeneralUtility::makeInstance(RepositoryDemandEvent::class, $demand, $queryBuilder);
-        $this->dispatcher->dispatch(get_class($this), $name, [$event]);
+        $this->dispatcher->dispatch(
+            GeneralUtility::makeInstance(RepositoryDemandEvent::class, $demand, $queryBuilder)
+        );
```

Comment:
Considering how the Demander extension has turned out and how the TYPO3 core is working, I'm thinking it could maybe be better to supply table names and aliases + ExpressionBuilder and return a restriction as a CompositeExpression. Check out https://github.com/pixelant/demander/pull/1/files#diff-c7693fa56dece674a5a3a16f48d8e59b4a8d08d8d344f246f00fa6d7ca24c229R20-R22

Reply:
Hmm, in this case, if fireDemandEvent() method will return CompositeExpression, it can't called like this, the idea is changed I think)

Reply:
Maybe it's better just return an event instead?

Reply:
Do what you think is best and then I can look at it again.